### PR TITLE
replace mintpasses with ipnft metadata signature validation

### PR DESCRIPTION
**Pull request by nour-karoui**

### CODE REVIEW

**Comment by elmariachi111** [comment]
please check why your formatter is reformatting everything and resubmit again. We're using forge as formatter (`forge fmt`). This should work out of the box.

**Comment by DorianWilhelm** [comment]
@nour-karoui Do you mind fixing your local formatting settings? It's quite hard to find the actual changes :D

File: src/IPNFT.sol [hunk 1]
```diff
@@ -140,6 +141,34 @@ contract IPNFT is ERC721URIStorageUpgradeable, ERC721BurnableUpgradeable, IReser
         return reservationId;
     }
 
+    function mintReservation(
```
Comment (elmariachi111) [inline src/IPNFT.sol:144]: replace the other function with this one. There are no mintpasses after this change.

File: src/Mintpass.sol [hunk 2]
```diff
@@ -82,6 +83,15 @@ contract Mintpass is AccessControl, ERC721BBaseTokenURI, ERC721BBurnable, IAutho
         return true;
     }
 
+    /// @dev see {IAuthorizeMints-isValidSignature}
+    /// @dev reverts when signature is not valid
+    /// @param signature must be a bytes value
+    /// @param _tokenURI must be a string value
+    function isValidSignature(bytes memory signature, string memory _tokenURI) external view override returns (bool) {
```
Comment (elmariachi111) [inline src/Mintpass.sol:90]: this function is fully unrelated to a mintpass. Instead, write a new Authorizer and adapt its interface to what you need. The existing one `function authorizeMint(address minter, address to, bytes memory data) external view returns (bool);` is totally sufficient to hand over signatures to the authorizer (you only must make sure to be able to reconstruct a signed message in the authorizer, but that can be virtually anything)

**Review by elmariachi111 — CHANGES_REQUESTED** [review]

File: src/Mintpass.sol [hunk 3]
```diff
@@ -4,7 +4,8 @@ pragma solidity ^0.8.17;
 import "erc721b/extensions/ERC721BBaseTokenURI.sol";
 import "erc721b/extensions/ERC721BBurnable.sol";
 import { AccessControl } from "@openzeppelin/contracts/access/AccessControl.sol";
-
+import { ECDSA } from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
```
Comment (elmariachi111) [inline src/Mintpass.sol:7]: make sure this file doesn't change at all since it is not upgradeable.

File: test/MintAuthorizer.t.sol [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    function setUp() public {
+        vm.startPrank(deployer);
+        authorizer = new SignatureMintAuthorizer(_signer);
+        vm.stopPrank();
+    }
+
+    function testAuthorizeMint() public {
+        vm.startPrank(deployer);
+        bytes memory data =
```
Comment (elmariachi111) [inline test/MintAuthorizer.t.sol:27]: don't use a fixed string. Create this signature inside the test.

**Review by elmariachi111 — CHANGES_REQUESTED** [review]

**Review by elmariachi111 — APPROVED** [review]

**Pull request merged**